include/misc/stack.h

```diff
@@ -90,8 +87,20 @@ static inline void stack_analyze(const char *name, const char *stack,
  * @param name Name of the stack
  * @param sym The symbol of the stack
  */
-#define STACK_ANALYZE(name, sym) \
-	stack_analyze(name, K_THREAD_STACK_BUFFER(sym), \
-		      K_THREAD_STACK_SIZEOF(sym))
+#if defined(CONFIG_INIT_STACKS)
+#define STACK_ANALYZE(name, sym)					\
+	do {								\
+		const char *stack = K_THREAD_STACK_BUFFER(sym);		\
+		size_t size = K_THREAD_STACK_SIZEOF(sym);		\
+									\
+		unsigned unused = stack_unused_space_get(stack, size);	\
+		unsigned pcnt = ((size - unused) * 100) / size;		\
+									\
+		LOG_INF("%s :\tunused %u\tusage %u / %u (%u %%)",	\
```

Comment:
there is a risk that somebody use that macro in file which does not have log module registered and `log.h` included at all. I would then prefer that to keep it as static inline function and use `LOG_MODULE_DECLARE(kernel)` to use known logger module.

Reply:
This is a header file so I cannot declare it for entire file as then it would conflict with possible module declaration from sources. If I will declare it inside the function will it correctly work?
I suppose it should but I don't see it used this way anywhere.

Wouldn't it be safer to simply direct all messages not logged under specific module to some default one?

Reply:
I used `LOG_MODULE_DECLARE`. The only problematic thing with it is that log level is for some reason incorrect unless I declare it explicitly.